app/uk/gov/hmrc/thirdpartyapplication/services/SubscriptionService.scala

```diff
@@ -84,6 +84,15 @@ class SubscriptionService @Inject()(applicationRepository: ApplicationRepository
       _ <- apiPlatformEventService.sendApiSubscribedEvent(app, apiIdentifier.context, apiIdentifier.version)
       _ <- auditSubscription(Subscribed, applicationId, apiIdentifier)
     } yield HasSucceeded
+  }  
+  
+  def createSubscriptionForApplicationMinusChecks(applicationId: UUID, apiIdentifier: APIIdentifier)(implicit hc: HeaderCarrier): Future[HasSucceeded] = {
+    for {
+      app <- fetchApp(applicationId)
+      _ <- subscriptionRepository.add(applicationId, apiIdentifier)
+      _ <- apiPlatformEventService.sendApiSubscribedEvent(app, apiIdentifier.context, apiIdentifier.version)
+      _ <- auditSubscription(Subscribed, applicationId, apiIdentifier)
+    } yield HasSucceeded
   }
```

Comment:
Should this also run this through he ApiFilter check to make sure you are allowed to subscribe?
The current rules for GK are that anyone can subscribe, _apart_ from Retired APIs (I think).

When this is called via DevHub it it will be very important that we check, as I expect this is what stops malicious posts into DevHub to subscribe to APIs you may not be able to subscribe to.

Reply:
It's not being called in devhub, the is GK specific and GK is doing the check for filtering.
That's the whole point of this work.

Reply:
Cool - I miss-understood. I was thinking about APM.